src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+import pencilImage from "../images/pencil.svg";
+import plusImage from "../images/plus.svg";
+import whitePencilIcon from "../images/White-Pencil.png";
+
+//
+let selectedCard;
+let selectedCardId;
+
+// Set initial image sources
+document.querySelector(".header__logo").src = logoImage;
+document.querySelector(".profile__avatar").src = avatarImage;
+document.querySelector(".profile__pencil-icon").src = pencilImage;
+document.querySelector(".profile__plus-icon").src = plusImage;
+document.querySelector(".white__pencil-icon").src = whitePencilIcon;
+
+// Initialize API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "df56a4ed-ea57-4811-a86a-44f3045b99eb",
+    "Content-Type": "application/json",
+  },
+});
+
+// Selectors
+const editModal = document.querySelector("#edit-modal");
+const cardModal = document.querySelector("#add-card-modal");
+const avatarModal = document.querySelector("#avatar-modal");
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = document.querySelector("#delete-form");
+const cardForm = document.querySelector("#card-form");
+const avatarForm = document.querySelector("#edit-avatar-form");
+const editProfileForm = document.querySelector("#edit-profile-form");
+const previewModal = document.querySelector("#preview-modal");
+
+// Open and close modal functions
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscClose);
+  modal.addEventListener("mousedown", handleOverlayClick);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscClose);
+  modal.removeEventListener("mousedown", handleOverlayClick);
+}
+
+// Escape key close handler
+function handleEscClose(event) {
+  if (event.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) closeModal(openedModal);
+  }
+}
+
+// Overlay click close handler
+function handleOverlayClick(event) {
+  if (event.target.classList.contains("modal_opened")) {
+    closeModal(event.target);
+  }
+}
+
+// Helper to set up modal close buttons
+function setUpCloseButtons() {
+  document.querySelectorAll(".modal__close-btn").forEach((button) => {
+    const modal = button.closest(".modal");
+    button.addEventListener("click", () => closeModal(modal));
+  });
+}
+setUpCloseButtons();
+
+// Load user info and cards
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    document.querySelector(".profile__avatar").src = userInfo.avatar;
+    document.querySelector(".profile__name").textContent = userInfo.name;
+    document.querySelector(".profile__description").textContent =
+      userInfo.about;
+
+    const cardsList = document.querySelector(".cards__list");
+    cards
+      .reverse()
+      .forEach((cardData) => cardsList.prepend(createCardElement(cardData)));
```

Comment:
**Could be improved**

You can make a universal function for adding a card into the section using different methods: `append`, `prepend`, or anything else:

```js

// The function accepts a card object and a method of adding to the section
// The method is initially `prepend`, but you can pass `append` 
function renderCard(item, method = "prepend") {
  
  const cardElement = getCardElement(item);
  // Add the card into the section using the method
  cardsList[ method ](cardElement);
}



```